CMakeLists.txt

```diff
@@ -58,6 +58,11 @@ endif()
 
 set (libs)
 
+include(CheckCXXCompilerFlag)
+check_cxx_compiler_flag (-Wformat HAVE_WFORMAT)
+if (HAVE_WFORMAT)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wformat")
```

Comment:
Generally its frowned upon to add global flags perhaps we can do this for the affected target(s) using 
[target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html)